### DISCUSSION TITLE
Improved support for Boolean type

### DISCUSSION THREAD

**Discussion by TonyGravagno** [araera111/mysql-to-zod#30] [Ideas]
MySQL provides BOOLEAN or BOOL as the synonym of TINYINT(1).  
In `toZod.ts`, only BIT is supported.

Zod supports z.boolean() and related coersion and nuances.

It will be helpful for the Zod schema to use z.boolean where we are certain that TINYINT(1) actually represents a Boolean.

On approval I will create and process an Issue.

**Comment by araera111** [comment]
I think tinyint(1) could store up to -128to127.
However, I know that in many cases it is treated as a bool.

If you get it with knex, it returns a number type.

I think this is something to consider about changing the type and schema when SELECT and (INSERT, UPDATE).

I know it doesn't happen very often, but what if you want to treat a value as a bool value in one table and a number value in another?
Such a case would be an exception.Let's not think about it.
I think it would be reasonable to make the option whether to treat tinyint(1) as a bool or not.

While I was responding to you, it was time for bed, so I am going to sleep today.
Sorry I didn't reply to all of them.

**Reply by TonyGravagno** [reply]
I appreciate all responses. I am patient and never in a hurry. This is a side project. We both have "real life" that takes priority. :)

Yes, the detail with TINYINT(1), and all such details are important. I will look closely to see if we can determine if the original CREATE TABLE used BOOLEAN, BOOL or BIT, so that we can distinguish between a synonym for TINYINT(1) versus a real TINYINT(1).
